backend/supabase/functions.sql

```diff
@@ -110,9 +110,13 @@ returns table (data jsonb, score real)
 immutable parallel safe
 language sql
 as $$
-  with recommended_contracts as (
-    select data, score
+  with recommendation_scores as materialized (
+    select contract_id, score
     from get_recommended_contract_scores_unseen(uid)
+    order by score desc
+  ), recommended_contracts as not materialized (
```

Comment:
I am not sure if you are familiar with `materialized` and `not materialized` but it does what it sounds like -- if you write `as materialized` then the CTE is computed and written to disk (useful if you want it to not repeat its work), if you write `as not materialized` then the CTE is not written to disk and re-queried when it's used later (useful if the work is cheap or if you want the planner to "see through" the CTE and choose how to do the query based on how it's used.) If you write nothing then the planner decides.

In this case, I want the recommendation scores to be written to disk, so they don't get computed twice, but I definitely don't want the `recommended_contracts` to be written to disk because (since I know that most contracts are valid) I know that's going to be way more contracts than I actually need. I want the planner to wait and combine this query with the `trending_contracts` and `new_contracts` filters and use the index I made.